src/CodeGen_LLVM.cpp

```diff
@@ -307,7 +314,12 @@ CodeGen_LLVM *CodeGen_LLVM::new_for_target(const Target &target,
         return make_codegen<CodeGen_PowerPC>(target, context);
     } else if (target.arch == Target::PNaCl) {
         return make_codegen<CodeGen_PNaCl>(target, context);
+#ifdef WITH_HEXAGON
```

Comment:
Looks like I was wrong. I think what's going on is:

- The makefile either sets -DWITH_FOO=1, or doesn't set it at all.
- The CMakeFile either set -DWITH_FOO, or doesn't set it at all

So they do indeed need to be ifdef, and the CMakefile needs to be fixed to only turn it on for llvm 3.9+